### PR TITLE
Feature/lt analytics project

### PR DESCRIPTION
**Pull request by sandinosaso**
This project specification is defined here:

https://github.com/lets-talk/letstalk-widget-js/issues/4

Screen Shot 2019-04-05 at 10 28 51 AM

For our wiki (so HR people can help clients configure Analytics):
(Spanish information explained)

How to set up a tracking ID:
https://support.google.com/analytics/answer/3123666
Advantages of having a tracking ID:
https://support.google.com/analytics/answer/3123663?hl=es&ref_topic=3123660
How session unification works:
https://support.google.com/analytics/answer/4574780?hl=es

### CODE REVIEW

File: packages/analytics/src/__tests__/analytics.spec.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+      it('Creates Analytics instance without crashing', () => {
+        const instance = new LTAnalytics(mockConfiguration);
+
+        expect(instance).not.toBeFalsy();
+      });
+
+      it('It creates a tracker for all defined tracking IDs', () => {
+        const instance = new LTAnalytics(mockConfiguration);
+
+        expect(instance).not.toBeFalsy();
```
Comment (alex-sysoev) [inline packages/analytics/src/__tests__/analytics.spec.ts]: Is this assertion necessary ?

File: packages/analytics/src/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  private shouldTrackEvent(trackerID: string, eventCategory: string) {
+    return this.options.events[trackerID][eventCategory] && this.options.events[trackerID][eventCategory].enabled;
+  }
+
+  private shouldTrackScreenView(trackerID: string, screenName: string) {
+    return this.options.screenViews[trackerID][screenName] && this.options.screenViews[trackerID][screenName].enabled;
+  }
+
+  public set(key: string, value: string) {
```
Comment (alex-sysoev) [inline packages/analytics/src/analytics.ts]: What this function actually sets ? Just to bother you ))) May be we could assign a bit more descriptive name to this function :cat2:

File: packages/analytics/src/analytics.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  constructor(options: ILTAnalyticsOptions) {
+    this.options = options;
+    this.allTrackersIDs = this.options.trackers;
+
+    if (!this.allTrackersIDs || this.allTrackersIDs.length < 1) {
+      throw new Error('Configuration Error. You must specify at least one tracker');
+    }
+
+    const defaultTrackingID = this.allTrackersIDs[0];
+    analytics.initialize(defaultTrackingID, options.createOptions);
```
Comment (dbarria) [inline packages/analytics/src/analytics.ts]: I think you are creating the same tracker twice. Once in `analytics.initialize` and again in `this.allTrackersIDs.forEach`

File: packages/analytics/src/analytics.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    this.allTrackersIDs.forEach((tracker) => {
+      analytics.create(tracker, this.getTrackerName(tracker));
```
[the remark this answers is not in the document]
Comment (sandinosaso) [reply packages/analytics/src/analytics.ts]: Agree with you on this

**Pull request merged**